### ISSUE TITLE
String literals in query not serialized correctly

### ISSUE THREAD

**Issue by arunisaac** [roelj/guile-sparql#5]
String literals in query get serialized without double quotes.

For example.
```
(let ((rdf (prefix "http://www.w3.org/1999/02/22-rdf-syntax-ns#"))
      (rdfs (prefix "http://www.w3.org/2000/01/rdf-schema#"))
      (internal (prefix "http://localhost/")))
  (select '(subject predicate object)
          `((subject predicate object)
            (subject ,(internal "property") "value"))))
```
gets serialized to
```
SELECT ?subject ?predicate ?object 
WHERE 
{
  ?subject ?predicate ?object .
  ?subject <http://localhost/property> value .
}
```
Notice the missing double quotes around `value`.

**Comment by roelj** [comment]
I can confirm the problem. As for solving it: I'm not sure what's the best way to go forward.
We could make the `prefix` procedure not return a string for the absolute URI, but that changes the output of `prefix`.

If you have any particular suggestions, I'd love to hear/read them. Otherwise, I'll tinker a bit and report back.

**Comment by arunisaac** [comment]
Yes, I think it's best that `prefix` does not output a string. IRIs should not be treated as strings. It is best to have separate data types for IRIs and literals. [guile-rdf](https://framagit.org/tyreunom/guile-rdf) might provide some inspiration.
